File: game_of_life/mrugger/java/GamePlayer.java


class GamePlayer implements IGridable
{
  private GameBoard current_board;
  private GameBoard next_board;

  public GamePlayer(GameBoard board)
  {
    current_board = board;
    Pair size = new Pair(current_board.get_height(), current_board.get_width());
    next_board = new GameBoard(size);
  }


  public void next_generation()
  {
    next_board.next_generation(current_board);
    GameBoard temp = current_board;
    current_board = next_board;
    next_board = temp;
  }


  // IGridable

  public int get_width()
  {
    return current_board.get_width();
  }

  public int get_height()
  {
    return current_board.get_height();
  }

  public CellType node_value(int x, int y)
  {
    return current_board.node_value(x, y);
  }

  public boolean in_range(int x, int y)
  {
    return current_board.in_range(x, y);
  }

  public boolean out_of_range(int x, int y)
  {
    return current_board.out_of_range(x, y);
  }
}

File: game_of_life/mrugger/ruby/game.player.rb
require './game.types.rb'
require './game.board.rb'


class GamePlayer

  private

  @grid_current
  @grid_next


  public

  def initialize(gameboard1, gameboard2)
    @grid_current = gameboard1
    @grid_next = gameboard2
  end
  

  # Player interface

  def next_generation
    @grid_next.next_generation(@grid_current)
    temp = @grid_current
    @grid_current = @grid_next
    @grid_next = temp
  end


  # Grid interface

  def get_width
    @grid_current.get_width
  end

  def get_height
    @grid_current.get_height
  end

  def node_value(x, y)
    @grid_current.node_value(x, y)
  end

  def in_range?(x, y)
    @grid_current.in_range?(x, y)
  end

  def out_of_range?(x, y)
    @grid_current.out_of_range?(x, y)
  end

  def for_each_cell
    @grid_current.for_each_cell do |x, y|
      yield x, y
    end
  end
end
